docs/pages/docs/guides/images-and-files.mdx

```diff
@@ -0,0 +1,272 @@
+import { Markdown } from '../../../components/Markdown';
+import { Well } from '../../../components/primitives/Well';
+import { RelatedContent } from '../../../components/RelatedContent';
+
+# Images and Files
+
+Keystone [field types](../apis/fields) contain both the `image` and `file`. These two field types can be used to reference, and if required, serve Images and/or Files from Keystone. These field types can be backed by either local or [Amazon S3 storage](https://aws.amazon.com/s3/).
+
+## Storage Options - S3 and Local
+
+Getting started with Images and Files in Keystone requires you first to create a `storage` object in your Keystone configuration file. Any number of stores can be set up within the `storage` object, and you can mix and match between `local` and `s3` depending on your use case.
+
+The `storage` object defines how and where the assets are stored and accessed by both Keystone and the client frontend. This object defines:
+
+- The `kind` of storage being used - either `s3` or `local`
+- The `type` of `field` the storage is being used for - either `file` or `image`
+- A function to generate the URL (`generateUrl`) that Keystone returns in the GraphQL API pointing to the location or the storage where the assets can be accessed
+- The actual location where Keystone stores the assets - either a local `path` or `s3` bucket details
+- The location Keystone will serve the assets from - either a `serverRoute` for `local` or a `proxied` connection for `s3`. Both of these options add a route to the Keystone backend which the files can be accessed from
+
+### Defining `storage` in Keystone config
+
+First, we are going to use [dotenv](https://www.npmjs.com/package/dotenv) to retrieve the S3 bucket and URL details from a `.env` file or set environment variables.
+
+In your Keystone Config file...
```

Comment:
```suggestion
In your Keystone Config file, pull in your environment variables and maps them to some easy to use names
```
In this case I think it's better to explain the intent before the code sample, as opposed to explaining in a callout underneath. This feels more consistent with similar code references in this guide and others.